### PR TITLE
kratos: all flows

### PR DESCRIPTION
**Pull request by elbotho**
Original PR: https://github.com/serlo/frontend/pull/1779

# TODOs
- [x] #1903
- [ ] some testing with german strings
- [ ] rocket chat: general oauth login flow
- [ ] rocket chat: it works in localhost, but not in staging. Problem identified (always remeber: in such cases, run to CF Worker!), Solution on the way
- [ ] rocket chat: register user that has no rocket chat acc yet

Done 🎉
- [x] https://github.com/serlo/frontend/issues/1886
- [x] https://github.com/serlo/frontend/issues/1887
- [x] #1902
- [x] check bundle size
- [x] todo comments
- [x] test image upload
- [x] test image upload with just registered user

### CODE REVIEW

File: .env.local [hunk 1]
```diff
@@ -10,3 +10,6 @@ HYDRA_HOST_STAGING=https://hydra.${NEXT_PUBLIC_SERLO_DOMAIN_STAGING}
 HYDRA_CLIENT_ID_LOCAL=frontend.serlo.org
 HYDRA_CLIENT_SECRET_LOCAL=frontend.serlo.org
 HYDRA_HOST_LOCAL=http://localhost:4444
+KRATOS_HOST_STAGING=https://kratos.${NEXT_PUBLIC_SERLO_DOMAIN_STAGING}
+KRATOS_HOST_LOCAL=http://localhost:4433
+ORY_SDK_URL="(NEXT_PUBLIC_ENV=local)?KRATOS_HOST_LOCAL:KRATOS_HOST_STAGING"
```
Comment (hugotiburtino) [inline .env.local]: ORY_SDK_URL is maybe unnecessary, check it

Comment (elbotho) [reply .env.local]: Seems to be unused. will remove it now, let's see if it breaks something :)

Comment (hugotiburtino) [reply .env.local]: Just for info: Its use is hidden in https://github.com/serlo/frontend/blob/ded405a75d9d924a0fc1880becbb46493bddfbd9/src/pages/api/.ory/%5B...paths%5D.ts
See https://github.com/ory/integrations#nextjs
But if you go to the [source code](https://github.com/ory/integrations/blob/main/src/next-edge/index.ts#L44-L58) you see that we already overwrite it

File: src/pages/api/.ory/[...paths].ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+const KRATOS_HOSTS = {
+  production: process.env.KRATOS_HOST_PRODUCTION,
+  staging: process.env.KRATOS_HOST_STAGING,
+  local: process.env.KRATOS_HOST_LOCAL,
+}
+
+const KRATOS_HOST = KRATOS_HOSTS[process.env.NEXT_PUBLIC_ENV || 'local']
+
+// TODO: this should probably be handled in CF Worker instead since it changes independent of Frontend.
```
Comment (elbotho) [inline src/pages/api/.ory/[...paths].ts:17]: I think this is fine for the foreseeable future. @hugotiburtino or do you have a solution in the making?

**Comment by hugotiburtino** [comment]
[Umzug Usermanagement](https://trello.com/c/XhyRVD6U/12-umzug-usermanagement)

**Pull request merged**